### ISSUE TITLE
RBF failure results in new, rejected txn id replacing old correctly broadcast one

### ISSUE THREAD

**Issue by gdvine** [spesmilo/electrum#3000]
Steps to reproduce:

1.  Create low-fee txn
2.  Copy txn id
3.  Increase fee to an amount that the addresses in the txn cannot accomodate, broadcast & get rejected by network
4.  Check txn id & see that a new one was generated for the invalid txn, and the old, valid txn id that successfully broadcast is no longer accessible.

**Comment by ecdsa** [comment]
I doubt this is real
can you reproduce that issue?

**Comment by gdvine** [comment]
I think this issue is different from what I have described, as the story continued differently. Here are the steps that happened:

1. Created low-fee txn in Electrum: stuck in mempool
2. Increased fee
3. Block explorer showed new txn ID as invalid (error not present)
4. Tried increasing fee _again_, got error on broadcast "Cannot bump fee: cound not find suitable outputs" (fixed that typo in wallet.py in my PR!), even though there was plenty of bitcoin in the wallet to pay a higher fee
5. Hours later, increased fee txn gets included in block & block explorer shows properly.

So perhaps the issue was actually with the block explorer not recognizing the RBF txn for some reason.

However, is the error that Electrum threw in step 4 expected behavior?
